client/src/features/ProjectPageV2/ProjectPageContent/CodeRepositories/CodeRepositoryDisplay.tsx

```diff
@@ -483,72 +486,54 @@ function RepositoryPermissions({ repositoryUrl }: RepositoryPermissionsProps) {
     repositoryProviderMatch?.repository_metadata?.permissions,
   ]);
 
-  const [isDetailsOpen, setIsDetailsOpen] = useState(false);
-  const toggleDetails = useCallback(() => {
-    setIsDetailsOpen((open) => !open);
-  }, []);
-
-  const commonButtonClasses = [
-    "border",
-    "rounded-circle",
-    "fs-6",
-    "lh-1",
-    "p-2",
-  ];
-
-  const buttonContent = isLoading ? (
-    <Loader className="bi" inline size={16} />
+  const badgeIcon = isLoading ? (
+    <Loader className="me-1" inline size={16} />
   ) : permissions.pull && permissions.push ? (
-    <CheckCircleFill className="bi" />
+    <CircleFill className={cx("me-1", "bi")} />
   ) : permissions.pull ? (
-    <SlashCircleFill className="bi" />
+    <CircleFill className={cx("me-1", "bi")} />
   ) : (
-    <XCircleFill className="bi" />
+    <CircleFill className={cx("me-1", "bi")} />
   );
 
-  const buttonClasses = isLoading
-    ? ["border-dark-subtle", "bg-light", "text-dark"]
-    : permissions.pull && permissions.push
-    ? ["border-success", "bg-success-subtle", "text-success"]
+  const badgeText = isLoading
+    ? ""
+    : permissions.push
+    ? "Push & pull"
     : permissions.pull
-    ? ["border-warning", "bg-warning-subtle", "text-warning"]
-    : ["border-danger", "bg-danger-subtle", "text-danger"];
+    ? "Pull only"
+    : "No access";
+
+  const badgeColorClasses = isLoading
+    ? ["border-dark-subtle", "bg-light", "text-dark-emphasis"]
+    : permissions.push
+    ? ["border-success", "bg-success-subtle", "text-success-emphasis"]
+    : permissions.pull
+    ? ["border-warning", "bg-warning-subtle", "text-warning-emphasis"]
+    : ["border-danger", "bg-danger-subtle", "text-danger-emphasis"];
 
   return (
-    <>
-      <Button
-        className={cx(...commonButtonClasses, ...buttonClasses)}
-        onClick={toggleDetails}
-        title="View repository permissions"
-      >
-        {buttonContent}
-        <span className="visually-hidden">View repository permissions</span>
-      </Button>
-      <Modal
-        size="lg"
-        fullscreen="sm"
-        isOpen={isDetailsOpen}
-        toggle={toggleDetails}
-        centered
-      >
-        <ModalHeader toggle={toggleDetails}>Repository permissions</ModalHeader>
-        <RepositoryPermissionsModalContent repositoryUrl={repositoryUrl} />
-        <ModalFooter>
-          <div className="d-flex justify-content-end">
-            <Button color="outline-danger" onClick={toggleDetails}>
-              <XLg className={cx("bi", "me-1")} />
-              Close
-            </Button>
-          </div>
-        </ModalFooter>
-      </Modal>
-    </>
+    <Badge pill className={cx("border", badgeColorClasses)}>
+      {badgeIcon}
+      <span className="fw-normal">{badgeText}</span>
```

Comment:
Avoid rendering empty nodes:

```suggestion
      {badgeText && (<span className="fw-normal">{badgeText}</span>)}
```